src/main/java/com/zaxxer/nuprocess/internal/BaseEventProcessor.java

```diff
@@ -48,6 +48,10 @@ public abstract class BaseEventProcessor<T extends BasePosixProcess> implements
    private CyclicBarrier startBarrier;
    private AtomicBoolean isRunning;
 
+   // avoid unnecessary malloc calls by reusing native pointer
+   // this field is thread safe as this is only used by Epoll/KQueue thread
+   protected final IntByReference tempIntPointer = new IntByReference();
```

Comment:
I don't like `temp` names for things that aren't temporary. This doesn't get allocated and then cleaned up; it gets reused over and over. I'd call it something like `exitCodePointer` or something like that.